### PR TITLE
Remove events rule from entity-operator

### PR DESCRIPTION
**Pull request by fvaleri**
This rule was only needed by the old implementation of the Topic Operator that has recently been removed.

### CODE REVIEW

**Review by scholzj — COMMENTED** [review]
What about `020-ClusterRole-strimzi-cluster-operator-role.yaml`? Shouldn't you remove the core API events from there as well? I think CO itself needs only the `events.k8s.io` IIRC.

File: cluster-operator/src/main/resources/cluster-roles/020-ClusterRole-strimzi-cluster-operator-role.yaml [hunk 1]
```diff
@@ -86,7 +86,6 @@ rules:
       - patch
       - update
   - apiGroups:
-      - "" # legacy core events api, used by topic operator
       - "events.k8s.io" # new events api, used by cluster operator
     resources:
       # The cluster operator needs to be able to create events and delegate permissions to do so
```
Comment (scholzj) [inline cluster-operator/src/main/resources/cluster-roles/020-ClusterRole-strimzi-cluster-operator-role.yaml:91]: Does it still delegate the events? Or are they now used only in the CO?

Comment (fvaleri) [reply cluster-operator/src/main/resources/cluster-roles/020-ClusterRole-strimzi-cluster-operator-role.yaml:91]: Both TO and UO do not create any event, so delegation is not required. Do you want me to update the comment?

Comment (scholzj) [reply cluster-operator/src/main/resources/cluster-roles/020-ClusterRole-strimzi-cluster-operator-role.yaml:91]: Yeah, I guess that would be perfect. Thanks.

**Review by scholzj — COMMENTED** [review]
The change looks good, but you need to run `make helm_install` once again 😉

**Review by scholzj — APPROVED** [review]

**Review by ppatierno — APPROVED** [review]
LGTM

**Comment by fvaleri** [comment]
I think failed tests are unrelated.

**Pull request merged**